file-share-service-desktop-client/App.xaml.cs

```diff
@@ -65,5 +85,44 @@ protected override void ConfigureModuleCatalog(IModuleCatalog moduleCatalog)
             moduleCatalog.AddModule<SearchUiModule>();
             moduleCatalog.AddModule<AdminUiModule>();
         }
+
+        private void SetupExceptionHandling(ILogger logger)
+        {
+            AppDomain.CurrentDomain.UnhandledException += (s, e) =>
+                LogUnhandledException((Exception)e.ExceptionObject, "AppDomain.CurrentDomain.UnhandledException", logger);
+
+            DispatcherUnhandledException += (s, e) =>
+            {
+                LogUnhandledException(e.Exception, "Application.Current.DispatcherUnhandledException", logger);
+                e.Handled = true;
+            };
+
+            TaskScheduler.UnobservedTaskException += (s, e) =>
+            {
+                LogUnhandledException(e.Exception, "TaskScheduler.UnobservedTaskException", logger);
+                e.SetObserved();
+            };
+        }
+
+        private void LogUnhandledException(Exception exception, string source, ILogger logger)
+        {
```

Comment:
Instead of having an `ILogger` parameter, we could instead get the logger from the DI container `var logger = Container.Resolve<ILogger<App>>();`
Doing this means we'll get the type name in the log message:
`2021-11-01 12:41:41 [Error] [] Unhandled exception in FileShareService.DesktopClient v1.0.0.0`
vs
`2021-11-01 12:37:39 [Error] [UKHO.FileShareService.DesktopClient.App] Unhandled exception in FileShareService.DesktopClient v1.0.0.0`

Reply:
Thank you for review comments. Implemented this change.